src/Routes/Filter/RouteFilterOptions.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mezzio\Tooling\Routes\Filter;
+
+use function get_object_vars;
+use function in_array;
+use function is_array;
+use function is_string;
+
+final class RouteFilterOptions
+{
+    private string $middleware;
+    private string $name;
+    private string $path;
+
+    /** @var array<array-key,string> */
+    private array $methods = [];
+
+    /**
+     * @param string|array<array-key,string> $methods
+     */
+    public function __construct(
```

Comment:
Generally speaking, I would prefer to see no default options here and instead unions of non-empty-string|null perhaps - that way it's explicit that null means the option is unset and should be ignored, and otherwise, you have a non-empty-string to work with.

Reply:
I don't quite follow. Do you mean something like:

```
public function __construct(
    private ?string $middleware = null,
    private ?string $name = null,
    private ?string $path = null,
    $methods = null
)
```